src/DatabaseGpt.Npgsql/NpgsqlDatabaseGptProvider.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System.Data;
+using System.Text;
+using Dapper;
+using DatabaseGpt.Abstractions;
+using DatabaseGpt.Abstractions.Exceptions;
+using Npgsql;
+
+namespace DatabaseGpt.Npgsql;
+
+public class NpgsqlDatabaseGptProvider : IDatabaseGptProvider, IDisposable
+{
+    private readonly NpgsqlDatabaseGptProviderConfiguration settings;
+    private bool disposedValue;
+
+    public NpgsqlDatabaseGptProvider(NpgsqlDatabaseGptProviderConfiguration settings)
+    {
+        this.settings = settings;
```

Comment:
The `settings` argument is used only in the constructor, so you don't need to store it in a private field.